### PR TITLE
Update insertionIndex code to deal with any type of note

### PR DESCRIPTION
**Pull request by dwertheimer**
git shows lots of changes (I guess due to Lint config), but I only made one change (line 89)

### CODE REVIEW

File: helpers/paragraph.js [hunk 1]
```diff
@@ -142,16 +138,12 @@ export function calcSmartPrependPoint(note: TNote): number {
  * @param {string} paraText - the text to prepend
  * @param {ParagraphType} paragraphType - the usual paragraph type to prepend
  */
-export function smartPrependPara(
-  note: TNote,
-  paraText: string,
-  paragraphType: ParagraphType,
-): void {
+export function smartPrependPara(note: TNote, paraText: string, paragraphType: ParagraphType): void {
```
Comment (jgclark) [inline helpers/paragraph.js:141]: When a func definition has more than a couple of parameters, following nmn's lead, I lay them out this way to make it easier to read.

File: helpers/paragraph.js [hunk 2]
```diff
@@ -173,12 +165,7 @@ export function findEndOfActivePartOfNote(note: TNote): number {
       cancelledHeaderLine = i
     }
   }
-  const endOfActive =
-    (doneHeaderLine > 0)
-      ? doneHeaderLine
-      : ((cancelledHeaderLine > 0)
-        ? cancelledHeaderLine
-        : lineCount)
+  const endOfActive = doneHeaderLine > 0 ? doneHeaderLine : cancelledHeaderLine > 0 ? cancelledHeaderLine : lineCount
```
Comment (jgclark) [inline helpers/paragraph.js:168]: I can't read nested ? : without layout structure.
I don't mind you removing brackets but please leave the layout.

**Review by jgclark — CHANGES_REQUESTED** [review]
Thanks for finding this, @dwertheimer.
Approved, bar the couple of layout comments below.

File: helpers/paragraph.js (hunk 1 above)
Comment (dwertheimer) [reply helpers/paragraph.js:141]: TBH, I didn't (intentionally) make these formatting changes, VSCode/Prettier/.eslint must have reformatted on save. I don't know how to tell it not to. Do you or @mikeerickson? Are you using the standard .eslintrc settings in the repo?

Comment (jgclark) [reply helpers/paragraph.js:141]: Aha.
I appear to have prettier turned off. I guess I got fed up of it changing things like this where occasionally I find particular layouts helpful.
I think I'm using the standard .eslintrc file.

**Comment by jgclark** [comment]
I've made the logic mods but left the layout as was.

**Pull request merged**